core/src/services/routing.js

```diff
@@ -114,21 +114,43 @@ class RoutingClass {
   }
 
   getHashPath(url = window.location.hash) {
+    // check for intent, if any
+    if (url && url.includes('?Intent=')) {
+      let hash = url.replace('#/#', '#'); // handle default hash and intent specific hash
+      const intentHash = RoutingHelpers.getIntentPath(hash.split('#')[1]);
+      if (intentHash) {
+        return intentHash;
+      }
+    }
+
     return url.split('#/')[1];
   }
 
   getModifiedPathname() {
+    // check for intent, if any
+    if (window.location.hash && window.location.hash.includes('?Intent=')) {
+      let hash = window.location.hash.replace('#/#', '').replace('#', '');
+      let intentPath = RoutingHelpers.getIntentPath(hash);
```

Comment:
Should this be `const intentPath`? It seems the identifier won’t be reassigned if I am not wrong.

Reply:
yep, good point 👍